### PR TITLE
7191 Must have a minimum of one share class

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
*Issue #:* /bcgov/entity#7191

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/services/filings/validations/annual_report.py [hunk 1]
```diff
@@ -37,8 +37,7 @@ def get_ar_dates(business: Business, start_date: datetime, next_ar_year):
     ar_min_date = datetime(next_ar_year, 1, 1).date()
 
     # Make sure min date is greater than or equal to last_ar_date or founding_date
-    if ar_min_date < start_date:
-        ar_min_date = start_date
+    ar_min_date = max(ar_min_date, start_date)
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/filings/validations/annual_report.py:40]: pylint was failing due to this if condition

File: legal-api/src/legal_api/services/filings/validations/common_validations.py [hunk 2]
```diff
@@ -18,6 +18,18 @@
...
 
 
+def has_at_least_one_share_class(filing_json, filing_type) -> Error:  # pylint: disable=too-many-branches
+    """Ensure that share structure contain at least 1 class by the end of the alteration or IA Correction filing."""
+    if filing_type in filing_json['filing'] and 'shareStructure' in filing_json['filing'][filing_type]:
+        share_classes = filing_json['filing'][filing_type] \
+            .get('shareStructure', {}).get('shareClasses', [])
+
+        if len(share_classes) == 0:
+            return 'A company must have a minimum of one share class.'
```
Comment (thorwolpert) [inline legal-api/src/legal_api/services/filings/validations/common_validations.py:28]: The return type doesn't match.

File: legal-api/src/legal_api/services/filings/validations/common_validations.py [hunk 3]
```diff
@@ -18,6 +18,18 @@
 from legal_api.utils.datetime import datetime as dt
 
 
+def has_at_least_one_share_class(filing_json, filing_type) -> Error:  # pylint: disable=too-many-branches
```
Comment (thorwolpert) [inline legal-api/src/legal_api/services/filings/validations/common_validations.py:21]: This should get wrapped by the type Optional,  and maybe be a str?

**Review by thorwolpert — CHANGES_REQUESTED** [review]
typing needs to be reviewed.

File: legal-api/src/legal_api/services/filings/validations/common_validations.py (hunk 3 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/services/filings/validations/common_validations.py]: fix

File: legal-api/src/legal_api/services/filings/validations/common_validations.py (hunk 2 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/services/filings/validations/common_validations.py:29]: fixed

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**